shared/actions/plan-billing.js

```diff
@@ -0,0 +1,208 @@
+// @flow
+
+import {call, put, select} from 'redux-saga/effects'
+import {takeLatest, delay} from 'redux-saga'
+import * as Constants from '../constants/plan-billing'
+import {apiserverGetRpcPromise, apiserverPostRpcPromise} from '../constants/types/flow-types'
+
+import type {UpdateBillingArgs, UpdateBilling, FetchBillingAndQuota, FetchBillingOverview, UpdateBillingAndQuota, UpdateAvailablePlans, BillingState, BootstrapData, UpdatePaymentInfo, BillingError} from '../constants/plan-billing'
+import type {SagaGenerator} from '../constants/types/saga'
+import type {TypedState} from '../constants/reducer'
+
+function updateBilling (updateBillingArgs: UpdateBillingArgs): UpdateBilling {
+  return {
+    type: Constants.updateBilling,
+    payload: updateBillingArgs,
+  }
+}
+
+function clearBillingError (): BillingError {
+  return {
+    type: Constants.billingError,
+    payload: undefined,
+  }
+}
+
+function fetchBillingAndQuota (): FetchBillingAndQuota {
+  return {
+    type: Constants.fetchBillingAndQuota,
+    payload: undefined,
+  }
+}
+
+function fetchBillingOverview (): FetchBillingOverview {
+  return {
+    type: Constants.fetchBillingOverview,
+    payload: undefined,
+  }
+}
+
+function bootstrapData (): BootstrapData {
+  return {
+    type: Constants.bootstrapData,
+    payload: undefined,
+  }
+}
+
+function apiArgsFormatter (args: Object) {
+  return Object.keys(args).map(key => {
+    return {key, value: args[key]}
+  })
+}
+
+function updateBillingArgsToApiArgs ({
+  planId,
+  cardNumber,
+  nameOnCard,
+  securityCode,
+  cardExpMonth,
+  cardExpYear,
+}: UpdateBillingArgs): Object {
+  return {
+    plan_id: planId,
+    cc_number: cardNumber.stringValue(),
+    cc_name: nameOnCard.stringValue(),
+    cc_cvc: securityCode.stringValue(),
+    cc_exp_month: cardExpMonth.stringValue(),
+    cc_exp_year: cardExpYear.stringValue(),
+  }
+}
+
+function * updateBillingSaga ({payload}: UpdateBilling): SagaGenerator<any, any> {
+  let planId = payload.planId
+  if (planId == null) {
+    const currentPlanIdSelector = ({planBilling: {plan}}: TypedState) => plan && plan.planId
+    planId = ((yield select(currentPlanIdSelector)): any)
+  }
+
+  // TODO (MM) some loading indicator: true
+  try {
+    yield call(apiserverPostRpcPromise, {
+      param: {
+        endpoint: 'account/billing_update',
+        args: apiArgsFormatter(updateBillingArgsToApiArgs({...payload, planId})),
```

Comment:
this is fine for now but we might want to just make a little wrapped version of this that does this for us already